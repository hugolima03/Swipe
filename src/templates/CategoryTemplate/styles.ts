import { CategoryInfo } from 'components/CategorySliderItem/styles'
import styled, { css } from 'styled-components'

export const ImageWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    position: relative;

    ${CategoryInfo} {
      border-radius: 0;

      p {
        font-size: calc(${theme.font.sizes.medium} * 2);
      }
    }
  `}
`
export const CategoryImage = styled.div`
  width: 100vw;

  height: 26rem;
  background: url('https://source.unsplash.com/user/willianjusten/1042x580');
`

export const ListWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    margin: 0 ${theme.spacings.xsmall};
    overflow: auto;
  `}
`
