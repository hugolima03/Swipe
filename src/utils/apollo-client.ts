import { ApolloClient, InMemoryCache } from '@apollo/client'

const client = new ApolloClient({
  uri: `${process.env.NEXT_PUBLIC_API_URL}`,
  cache: new InMemoryCache(),
  headers: { Authorization: `Bearer ${process.env.NEXT_PUBLIC_GRAPHCMS_TOKEN}` }
})

export default client
