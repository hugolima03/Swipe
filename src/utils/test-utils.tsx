import { MockedProvider, MockedResponse } from '@apollo/client/testing'

import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'

type CustomRenderProps = Omit<RenderOptions, 'queries'>

const mocks: MockedResponse<Record<string, unknown>>[] = []

const customRender = (
  ui: React.ReactElement,
  { ...renderOptions }: CustomRenderProps = {}
) =>
  render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <ThemeProvider theme={theme}>{ui}</ThemeProvider>
    </MockedProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
