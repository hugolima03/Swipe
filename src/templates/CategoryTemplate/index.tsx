import * as S from './styles'

const CategoryTemplate = () => (
  <S.Wrapper>
    <h1>CategoryTemplate</h1>
  </S.Wrapper>
)

export default CategoryTemplate
