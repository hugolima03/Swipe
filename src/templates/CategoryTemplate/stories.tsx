import { Story, Meta } from '@storybook/react'
import CategoryTemplate from '.'

export default {
  title: 'CategoryTemplate',
  component: CategoryTemplate
} as Meta

export const Default: Story = () => <CategoryTemplate />
