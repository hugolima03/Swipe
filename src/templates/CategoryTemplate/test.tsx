import { render, screen } from 'utils/test-utils'

import CategoryTemplate from '.'

describe('<CategoryTemplate />', () => {
  it('should render the heading', () => {
    const { container } = render(<CategoryTemplate />)

    expect(
      screen.getByRole('heading', { name: /CategoryTemplate/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
