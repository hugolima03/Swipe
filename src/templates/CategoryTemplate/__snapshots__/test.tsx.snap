// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryTemplate /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    CategoryTemplate
  </h1>
</main>
`
